app/views/csvfiles/new.html.erb

```diff
@@ -0,0 +1,155 @@
+            $.ajax({
+                url: '/graph/prev_file',
+                type: 'get',
+                data: { uid: <%= current_user.id %> },
+                success: function(data){
+                    displayPreviousFiles(data);
+                },
+                error: function(XMLHttpRequest, textStatus, errorThrown){
+                        console.log(errorThrown);
+                        let divAlert = document.createElement('div');
+                        divAlert.classList.add('alert');
+                        divAlert.classList.add('alert-danger');
+                        divAlert.innerHTML = 'There was some error in retrieving the file.';
+                        headerContainer.appendChild(divAlert);
+                }
+            });
+        });
+
+        function displayPreviousFiles(data){
+            var table=document.createElement('table');
+            for (let i = 0;i<data.length;i++){
+                let tr = document.createElement('tr');
+                let td = document.createElement('td');
+                let radio = document.createElement('input');
+                radio.type = 'radio';
+                radio.value = i;
+                radio.name = data[0]['csvfile']['uid'] + 'user_id';
+                td.appendChild(radio);
+                td.appendChild(document.createTextNode(data[i]['csvfile']['filetitle']));
+                tr.appendChild(td);
+                table.appendChild(tr);
+            }
+            let div = document.getElementById(SimpleDataGrapherObject.view.upload_button_container);
+            div.appendChild(table);
+            selectFile(data);
+        }
+        function selectFile(data){
+            $('#' + SimpleDataGrapherObject.view.uploadButtonId).click(function(){
+                let name = data[0]['csvfile']['uid'] + 'user_id';
+                let index = $('input[name='+ name +']:checked').val();
+                let allfiles = JSON.parse(data[index]['csvfile']['filepath']);
+                SimpleDataGrapherObject.view.usingPreviouslyUploadedFile();
+                SimpleDataGrapherObject.view.csvParser.completeCsvMatrix = allfiles['completeCsvMatrix'];
+                SimpleDataGrapherObject.view.csvParser.csvHeaders = allfiles['csvHeaders'];
+                SimpleDataGrapherObject.view.csvParser.csvSampleData = allfiles['csvSampleData'];
+                SimpleDataGrapherObject.view.csvParser.csvValidForYAxis = allfiles['csvValidForYAxis'];
+                SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose = allfiles['completeCsvMatrixTranspose'];
+                SimpleDataGrapherObject.view.continueViewManipulation('prevfile');
+            }); 
+        }
+        $('#' + SimpleDataGrapherObject.view.elementId + '_publish').click(function(){
+            console.log(SimpleDataGrapherObject.view.plotlyjsPlotter.dataHash);
+            console.log(SimpleDataGrapherObject.view.plotlyjsPlotter.graphType);
```

Comment:
are these required?

Reply:
I will remove the log statements in a follow up PR

Reply:
If the console statements aren't required then they should be removed in this PR itself. A PR should be complete for its purpose along with the most optimality reached till merge point by mentor and peer reviewing.
If it needs to be anyhow removed later and there is no use for it till the follow up PR, then these statements will just bloat the codebase, which isn't good in anycase.
I think these statements aren't required but still I'll wait for other's opinion on this.
CC: @CleverFool77 @jywarren @gauravano

Reply:
Oh, lemme take a look at them and remove the ones that aren't required!